engine/src/cmd/energetic.cpp

```diff
@@ -0,0 +1,197 @@
+        if ( maxenergy <= damageable->totalShieldEnergyCapacitance( shield ) )
+            return 0;
+        return ( (float) energy )/( maxenergy-damageable->totalShieldEnergyCapacitance( shield ) );
+    } else {
+        return ( (float) energy )/maxenergy;
+    }
+}
+
+//Returns the current ammt of energy left
+float Energetic::energyRechargeData() const
+{
+    return recharge;
+}
+
+float Energetic::fuelData() const
+{
+    return fuel;
+}
+
+float Energetic::getFuelUsage( bool afterburner )
+{
+    static float normalfuelusage = GameConfig::GetVariable( "physics", "FuelUsage", 1.0f );
+    static float abfuelusage     = GameConfig::GetVariable( "physics", "AfterburnerFuelUsage", 4.0f );
+    if (afterburner)
+        return abfuelusage;
+    return normalfuelusage;
+}
+
+float Energetic::getWarpEnergy() const
+{
+    return warpenergy;
+}
+
+void Energetic::increaseWarpEnergy( bool insys, float time )
+{
+    static bool WCfuelhack = GameConfig::GetVariable( "physics", "fuel_equals_warp", false );
+    if (WCfuelhack)
+        this->warpenergy = this->fuel;
+    this->warpenergy += (insys ? jump.insysenergy : jump.energy)*time;
+    if (this->warpenergy > this->maxwarpenergy)
+        this->warpenergy = this->maxwarpenergy;
+    if (WCfuelhack)
+        this->fuel = this->warpenergy;
+}
+
+float Energetic::maxEnergyData() const
+{
+    return maxenergy;
+}
+
+void Energetic::rechargeEnergy()
+{
+    static bool reactor_uses_fuel = GameConfig::GetVariable( "physics", "reactor_uses_fuel", false );
+    if ( (!reactor_uses_fuel) || (fuel > 0) )
+        energy += recharge * simulation_atom_var;
+}
+
+bool Energetic::refillWarpEnergy()
+{
+    static bool WCfuelhack = GameConfig::GetVariable( "physics", "fuel_equals_warp", false );
+    if (WCfuelhack)
+        this->warpenergy = this->fuel;
+    float tmp = this->maxwarpenergy;
+    if (tmp < this->jump.energy)
+        tmp = this->jump.energy;
+    if (tmp > this->warpenergy) {
+        this->warpenergy = tmp;
+        if (WCfuelhack)
+            this->fuel = this->warpenergy;
+        return true;
+    }
+    return false;
+}
+
+void Energetic::setAfterburnerEnergy( float aft )
+{
+    afterburnenergy = aft;
+}
+
+void Energetic::setEnergyRecharge( float enrech )
+{
+    recharge = enrech;
+}
+
+void Energetic::setFuel( float f )
+{
+    fuel = f;
+}
+
+
+
+void Energetic::setMaxEnergy( float maxen )
+{
+    maxenergy = maxen;
+}
+
+float Energetic::shieldRechargeData() const
+{
+    const Damageable *damageable = reinterpret_cast<const Damageable*>(this);
+    const Shield shield = damageable->shield;
+    return shield.recharge;
+}
+
+
+
+float Energetic::warpCapData() const
+{
+    return maxwarpenergy;
+}
+
+
+float Energetic::warpEnergyData() const
+{
+    if (maxwarpenergy > 0)
+        return ( (float) warpenergy )/( (float) maxwarpenergy );
+    if (jump.energy > 0)
+        return ( (float) warpenergy )/( (float) jump.energy );
+    return 0.0f;
+}
+
+
+
+
```

Comment:
can we please remove the dead space at the end of the file?

Reply:
Done.